src/Domain/ValueObjects/Wallet.cs

```diff
@@ -29,15 +15,16 @@ public bool HasBalance(decimal amount)
 
     public bool TryTranfer(Wallet other, decimal amount)
     {
```

Comment:
validate that the decimal amount we are checking against is a non-zero positive decimal, throw argument exception if something is wrong